nbgrader/converters/base.py

```diff
@@ -248,6 +260,24 @@ def set_permissions(self, assignment_id, student_id):
         for dirname, _, filenames in os.walk(dest):
             for filename in filenames:
                 os.chmod(os.path.join(dirname, filename), permissions)
+            # If groupshared, set dir permissions - see comment below.
+            if self.groupshared and os.stat(dirname).st_uid == os.getuid():
+                #for subdirname in subdirnames:
+                #    subdirname = os.path.join(dirname, subdirname)
+                try:   os.chmod(dirname, (os.stat(dirname).st_mode|0o2770) & 0o2777)
+                except PermissionError: pass
```

Comment:
Should these `pass` statements on the PermissionError raise an error? Presumably they  happen if the user doesn't actually  have permission to set this, which means they shouldn't be using nbgrader either maybe?

Reply:
Solved/commented on down below.  Now is more clever about changing permissions and logs a message.